### PR TITLE
XW-4262 | Load all data from liftigniter service to limit number of r…

### PR DESCRIPTION
**Pull request by hakubo**
…equests made to service

https://github.com/Wikia/pandora/pull/3432

### CODE REVIEW

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 1]
```diff
@@ -174,17 +201,17 @@ private static function isProduction() {
...
-	 * @param $metaDataFromService is actual Data returned by Liftigniter Metadata Service
+	 * @param Item $metaDataFromService is actual Data returned by Liftigniter Metadata Service
 	 *
 	 * @return bool
 	 */
-	private static function shoudlNoIndex( $metaDataFromService ) {
+	private static function shouldNoIndex( $metaDataFromService ) {
 		global $wgDisableShowInRecirculation;
 
 		return self::isPrivateOrNotProduction() ||
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php:206]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

[project-id]: wikia:app

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 2]
```diff
@@ -81,7 +82,7 @@ public static function isCorrectPageType() {
 		$isInShowableNamespaces = $title->exists() && $title->inNamespaces( $showableNamespaces );
 
 		if ( $isInShowableNamespaces && !WikiaPageType::isActionPage() &&
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: MINOR Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L28](https://github.com/Wikia/app/blob/f11e65cdf227d93a43c069ecd8384fe12783ca7c/extensions/wikia/Recirculation/RecirculationHooks.class.php#L28): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L41](https://github.com/Wikia/app/blob/f11e65cdf227d93a43c069ecd8384fe12783ca7c/extensions/wikia/Recirculation/RecirculationHooks.class.php#L41): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [RecirculationHooks.class.php#L110](https://github.com/Wikia/app/blob/f11e65cdf227d93a43c069ecd8384fe12783ca7c/extensions/wikia/Recirculation/RecirculationHooks.class.php#L110): Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L40](https://github.com/Wikia/app/blob/f81dc86413120a30deba8a33552076099b90ffcd/extensions/wikia/Recirculation/RecirculationHooks.class.php#L40): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [RecirculationHooks.class.php#L105](https://github.com/Wikia/app/blob/f81dc86413120a30deba8a33552076099b90ffcd/extensions/wikia/Recirculation/RecirculationHooks.class.php#L105): Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]
You will have to re-generate all Swagger generated code with the new version of Swagger Codegen

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 3]
```diff
@@ -174,17 +196,17 @@ private static function isProduction() {
 	}
 
 	/**
-	 * @param $metaDataFromService is actual Data returned by Liftigniter Metadata Service
+	 * @param Item $metaDataFromService is actual Data returned by Liftigniter Metadata Service
 	 *
 	 * @return bool
 	 */
-	private static function shoudlNoIndex( $metaDataFromService ) {
+	private static function shouldNoIndex( $metaDataFromService ) {
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php:203]: can we change this name (and logic withing this function) to `shouldIndex`?

**Review by rybmat — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L40](https://github.com/Wikia/app/blob/b273bcec5a564df805b8d2d14d1f7a831adb7903/extensions/wikia/Recirculation/RecirculationHooks.class.php#L40): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [RecirculationHooks.class.php#L105](https://github.com/Wikia/app/blob/b273bcec5a564df805b8d2d14d1f7a831adb7903/extensions/wikia/Recirculation/RecirculationHooks.class.php#L105): Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by latata — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L40](https://github.com/Wikia/app/blob/0f451c6acb828275b5741f94beff159e0375fa94/extensions/wikia/Recirculation/RecirculationHooks.class.php#L40): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [RecirculationHooks.class.php#L105](https://github.com/Wikia/app/blob/0f451c6acb828275b5741f94beff159e0375fa94/extensions/wikia/Recirculation/RecirculationHooks.class.php#L105): Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecirculationHooks.class.php#L40](https://github.com/Wikia/app/blob/53ee5f81f645b3454985a188acbdcf85cf23dbc0/extensions/wikia/Recirculation/RecirculationHooks.class.php#L40): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [RecirculationHooks.class.php#L105](https://github.com/Wikia/app/blob/53ee5f81f645b3454985a188acbdcf85cf23dbc0/extensions/wikia/Recirculation/RecirculationHooks.class.php#L105): Replace this "if-then-else" statement by a single "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1126)

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**